### PR TITLE
Configuration file

### LINKED ISSUE #129: chore: configuration file
### Background
It is frustrating, and error prone to get the configuration for the watch-tower correct given the index-specific configuration.

### Details
All _network_ CLI options should be removed and replaced with a mandatory configuration file config option. This should allow multiple networks to be configured so that users may run a sovereign watch-tower.

### Acceptance criteria

- [x] JSON configuration file for network configuration
- [x] CLI options cleaned up
- [ ] cowprotocol/dappnodepackage-cow-watch-tower#6

### PR DESCRIPTION
**Pull request by mfw78**
# Description

Configuration methodology (via CLI options) has been outgrown with the number of options. This PR removes all chain / network related configuration and moves to a configuration file.

# Changes

- [x] Move all configuration for chains to a dedicated file
- [x] Remove unused features (replay tx / replay block)
- [x] Eliminates dynamic download of filter policies (now within configuration file)
- [x] Update `README.md`

## How to test
1. Adjust the `config.json.example` for suitable configuration
2. Run and verify sync process completes warmup

## Related Issues

Fixes #129

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -343,6 +162,30 @@ function parseAddressOption(option: string, previous: string[] = []): string[] {
...
 
+function validateJSON(filePath: string): Config {
+  const ajv = new Ajv();
+  addFormats(ajv);
+  const schema: JSONSchemaType<Config> = JSON.parse(
+    fs.readFileSync(CONFIG_SCHEMA_PATH, "utf8")
+  );
+  const validate = ajv.compile(schema);
+  try {
+    // Read and prase the JSON file
```
Comment (alfetopito) [inline src/index.ts:173]:
```suggestion
    // Read and parse the JSON file
```

File: src/types/types.d.ts
Comment (alfetopito) [inline src/types/types.d.ts:1]: Nitpick: shouldn't this be also within the `generated/` folder?

File: src/utils/filterPolicy.ts [hunk 2]
```diff
@@ -52,35 +56,16 @@ export class FilterPolicy {
...
-      );
-    }
-    const config = await configResponse.json();
-    return {
-      defaultAction: config.defaultAction,
-      owners: new Map(Object.entries(config.owners)),
-      handlers: new Map(Object.entries(config.handlers)),
-    };
+  private convertToMap(object?: {
+    [k: string]: "ACCEPT" | "DROP" | "SKIP";
```
Comment (alfetopito) [inline src/utils/filterPolicy.ts:60]: Could this be somehow be reused from the types generated from the json schema?

File: src/types/config.schema.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    "filterPolicy": {
+                        "type": "object",
+                        "properties": {
+                            "defaultAction": {
+                                "type": "string",
+                                "enum": [
+                                    "ACCEPT",
+                                    "DROP",
+                                    "SKIP"
+                                ]
```
Comment (alfetopito) [inline src/types/config.schema.json:39]: Nitpick: create a definition for this type/enum to avoid repetition

**Review by alfetopito — APPROVED** [review]
Nice refactor!
Some nitpicks.

**Review by shoom3301 — APPROVED** [review]

File: src/types/config.schema.json (hunk 3 above)
Comment (mfw78) [reply src/types/config.schema.json]: Nice, done!

File: src/types/types.d.ts
Comment (mfw78) [reply src/types/types.d.ts:1]: At the moment, `generated/` is reserved for `typechain` generated artifacts

File: src/utils/filterPolicy.ts (hunk 2 above)
Comment (mfw78) [reply src/utils/filterPolicy.ts]: Done!

**Review by anxolin — COMMENTED** [review]
Half way in my review, i need to try it more. I send a few comments.

In general I feel that given the size of the change:
- you could have waited a bit for the review, is good that for such a change we have more eyes
- it looks like this PR does a bunch of things that would be better to keep in separate PRs


I like the config. Feels much better

File: README.md [hunk 4]
```diff
@@ -1,21 +1,21 @@
-# Watch-Tower for Composable CoWs 🐮🎶
+# Watch-Tower for Programmatic Orders 🐮🎶
 
 ## Overview
 
-The [`ComposableCoW`](https://github.com/cowprotocol/composable-cow) conditional order framework requires a watch-tower to monitor the blockchain for new orders, and to post them to the [CoW Protocol `OrderBook` API](https://api.cow.fi/docs/#/). The watch-tower is a standalone application that can be run locally as a script for development, or deployed as a docker container to a server, or dappnode.
+The [programmatic order](https://docs.cow.fi/cow-protocol/concepts/order-types/programmatic-orders) framework requires a watch-tower to monitor the blockchain for new orders, and to post them to the [CoW Protocol `OrderBook` API](https://docs.cow.fi/cow-protocol/reference/apis/orderbook). The watch-tower is a standalone application that can be run locally as a script for development, or deployed as a docker container to a server, or dappnode.
```
[the remark this answers is not in the document]
Comment (mfw78) [reply README.md:5]: While nice, the idea is mostly that there is a hyperlink there that explains it, and reduces duplication of typing / explanation.

File: src/commands/index.ts [hunk 5]
```diff
@@ -1,5 +1,2 @@
 export * from "./run";
-export * from "./runMulti";
-export * from "./replayBlock";
-export * from "./replayTx";
```
[the remark this answers is not in the document]
Comment (mfw78) [reply src/commands/index.ts:4]: 1. `runMulti` has been superceded (actually, `run` was removed and `runMulti` was repurposed to `run`).
2. `replayBlock` and `replayTx` weren't actually implemented (so removed dead stubs).

File: README.md [hunk 6]
```diff
@@ -1,21 +1,21 @@
-# Watch-Tower for Composable CoWs 🐮🎶
+# Watch-Tower for Programmatic Orders 🐮🎶
```
[the remark this answers is not in the document]
Comment (mfw78) [reply README.md:1]: Done :+1:

File: config.json.example [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+    "networks": [
```
[the remark this answers is not in the document]
Comment (mfw78) [reply config.json.example:2]: We could, I don't have a strong opinion either way on this, and only did this way so as to minimise work, and minimise type duplication between the `watch-tower` and the Pulumi configuration for infrastructure (not a strong argument though).

File: src/types/config.schema.json [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+                    "pageSize": {
+                        "type": "integer"
+                    },
+                    "filterPolicy": {
+                        "type": "object",
+                        "properties": {
+                            "defaultAction": {
+                                "$ref": "#/$defs/filterAction"
+                            },
+                            "owners": {
```
Comment (anxolin) [inline src/types/config.schema.json:36]: owners and handlers have the same type. would it be worth it to reference it? 

Actually would be probably be nice to define the FilterPolicy too. This way you will get a Typescript definition for it you can reuse, so you don't need to do things like:
` constructor(config: Config["networks"][number]["filterPolicy"]) {`

**Review by anxolin — COMMENTED** [review]
Worked great! 

soft approve

File: src/types/config.schema.json [hunk 9]
```diff
@@ -0,0 +1,79 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
```
[the remark this answers is not in the document]
Comment (mfw78) [reply src/types/config.schema.json:4]: Not sure what you mean by this?

**Pull request merged**